metalearners/metalearner.py

```diff
@@ -1123,6 +1123,55 @@ def _default_scoring() -> Scoring:
             return default_scoring
         return dict(default_scoring) | dict(scoring)
 
+    @property
+    def init_args(self) -> dict[str, Any]:
+        """Create initiliazation parameters for a new MetaLearner.
+
+        Importantly, this does not copy further internal state, such as the weights or
+        parameters of trained base models.
+        """
+        return {
+            "is_classification": self.is_classification,
+            "n_variants": self.n_variants,
+            "nuisance_model_factory": {
+                k: v
+                for k, v in self.nuisance_model_factory.items()
+                if k != PROPENSITY_MODEL
+                if k not in self._prefitted_nuisance_models
+            },
+            "treatment_model_factory": self.treatment_model_factory,
+            "propensity_model_factory": (
+                self.nuisance_model_factory.get(PROPENSITY_MODEL)
+                if PROPENSITY_MODEL not in self._prefitted_nuisance_models
+                else None
+            ),
+            "nuisance_model_params": {
+                k: v
+                for k, v in self.nuisance_model_params.items()
+                if k != PROPENSITY_MODEL
+                if k not in self._prefitted_nuisance_models
+            },
+            "treatment_model_params": self.treatment_model_params,
+            "propensity_model_params": (
+                self.nuisance_model_params.get(PROPENSITY_MODEL)
+                if PROPENSITY_MODEL not in self._prefitted_nuisance_models
+                else None
+            ),
+            "fitted_nuisance_models": {
+                k: v
```

Comment:
```suggestion
                k: deepcopy(v)
```
I think we should do a deep copy of the prefitted estimators as we did in the `__init__` method, see [this line](https://github.com/Quantco/metalearners/blob/2f428fbeb63be38443e404edea51e4a862b4e2fd/metalearners/metalearner.py#L504).

The idea behind it was that if the user reuses a model from another metalearner and refits this previous metalearner then the base model in the second metalearner is not refitted. In this case it should not happen as the model was already deep copied the first time it was reused but I think it would be better to have completely decoupled metalearners. Wdyt?

Reply:
Great catch!